test/adapters/integration.rb

```diff
@@ -235,17 +235,26 @@ def adapter_options
       end
 
       def create_connection(options = {})
-        if adapter == :default
-          builder_block = nil
-        else
-          builder_block = Proc.new do |b|
-            b.request :multipart
-            b.request :url_encoded
-            b.adapter adapter, *adapter_options
+        builder_block =
+          if adapter == :default
+            nil
+          else
+            Proc.new do |b|
+              if options[:proxy]
+                b.request :proxy, options[:proxy]
+              else
+                b.request :proxy
+              end
```

Comment:
Perhaps this part will just fade away when _all_ connections will have the Request::Proxy middleware?